### PR TITLE
[DO-1797]Storing property rights

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: business_register/converter/declaration.py [hunk 1]
```diff
@@ -27,13 +30,113 @@ def __init__(self):
...
+            type = TYPES.get(data.get('ownershipType'))
+            share = data.get('percent-ownership')
+            if share and share not in self.NO_DATA:
+                share = float(share.replace(',', '.'))
+            else:
+                share = None
+            owner_info = data.get('rightBelongs')
+            pep = None
+            # TODO: store value from ENIGMA
+            if owner_info and owner_info not in self.NO_DATA and owner_info not in ENIGMA:
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:73]: what if we add `None`, `''` to `self.NO_DATA`?

File: business_register/converter/declaration.py [hunk 2]
```diff
@@ -27,13 +30,113 @@ def __init__(self):
...
+            'Спільна власність': PropertyRight.JOINT_OWNERSHIP,
+            'Спільна сумісна власність': PropertyRight.COMMON_PROPERTY,
+            'Оренда': PropertyRight.RENT,
+            'Інше право користування': PropertyRight.OTHER_USAGE_RIGHT,
+            'Власником є третя особа': PropertyRight.OWNER_IS_ANOTHER_PERSON,
+            ('Право власності третьої особи, але наявні ознаки відповідно до частини 3 статті 46 '
+             'ЗУ «Про запобігання корупції»'): PropertyRight.BENEFICIAL_OWNERSHIP,
+            "[Член сім'ї не надав інформацію]": PropertyRight.NO_INFO_FROM_FAMILY_MEMBER,
+        }
+        ENIGMA = ['1', 'j', 'Інша особа (фізична або юридична)']
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:62]: We are waiting for an answer from NACP

File: business_register/converter/declaration.py [hunk 3]
```diff
@@ -27,13 +30,113 @@ def __init__(self):
...
+            pep = None
+            # TODO: store value from ENIGMA
+            if owner_info and owner_info not in self.NO_DATA and owner_info not in ENIGMA:
+                pep = Pep.objects.filter(nacp_id=int(owner_info)).first()
+            other_owner_info = data.get('rights_id')
+            if not pep and other_owner_info and other_owner_info not in ENIGMA:
+                pep = Pep.objects.filter(nacp_id=int(other_owner_info)).first()
+            additional_info = data.get('otherOwnership')
+            if not additional_info:
+                additional_info = ''
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:80]: Thank you, changed

File: business_register/converter/declaration.py (hunk 1 above)
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:73]: good idea

Comment (IegorSoboliev) [reply business_register/converter/declaration.py:73]: added

Comment (roman-tiukh) [reply business_register/converter/declaration.py]: empty string? (`''`)

File: business_register/converter/declaration.py [hunk 4]
```diff
@@ -207,16 +324,20 @@ def save_or_update_declaration(self):
                     f'cannot find declarations of the PEP with nacp_declarant_id: {nacp_declarant_id}'
                 )
                 continue
+
             pep = self.only_peps[nacp_declarant_id]
             for declaration_data in declarations_data:
+                declaration_type = declaration_data['declaration_type']
+                # TODO: predict storing changes from the declarant
+                if declaration_type not in [1, 2, 3, 4]:
```
Comment (roman-tiukh) [inline business_register/converter/declaration.py:332]: we should use constants here, not `1, 2, 3, 4`

File: business_register/converter/declaration.py (hunk 1 above)
Comment (IegorSoboliev) [reply business_register/converter/declaration.py]: I haven`t seen it yet

File: business_register/converter/declaration.py (hunk 4 above)
Comment (IegorSoboliev) [reply business_register/converter/declaration.py:332]: this is interim measure)

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**